src/lib/solana/index.ts

```diff
@@ -96,92 +112,111 @@ export class SolanaUtils {
-    } catch (error: unknown) {
-      console.error('TEST Transaction error:', error);
-      if (error instanceof Error) {
-        // Handle insufficient funds error
-        if (error.toString().includes('insufficient lamports')) {
-          throw new Error(
-            `Insufficient balance. Please make sure you have enough SOL to cover the transaction.`,
-          );
-        }
-        // Handle other known errors
-        if (error.toString().includes('Transaction simulation failed')) {
-          throw new Error(`Transaction failed. Please try again.`);
+      try {
+        // Create transaction
+        const transaction = new Transaction();
+        transaction.feePayer = fromPubkey;
+
+        // Create transfer instruction
+        const transferInstruction = SystemProgram.transfer({
+          fromPubkey,
+          toPubkey,
+          lamports: requiredAmount,
+        });
+
+        // Create Memo instruction
+        const memoInstruction = new TransactionInstruction({
+          keys: [{ pubkey: fromPubkey, isSigner: true, isWritable: true }],
+          programId: new PublicKey(MEMO_PROGRAM_ID),
+          data: Buffer.from(memo, 'utf-8'),
+        });
+
+        transaction.add(transferInstruction);
+        transaction.add(memoInstruction);
+
+        // Get latest blockhash
+        const { blockhash } =
+          await this.connection.getLatestBlockhash('confirmed');
+        transaction.recentBlockhash = blockhash;
+
+        // Sign transaction
+        const signedTransaction =
+          await connectedSolanaWallet.signTransaction(transaction);
+
+        // Send transaction and return signature immediately
+        const signature = await this.connection.sendRawTransaction(
+          signedTransaction.serialize(),
+          {
+            skipPreflight: false,
+            maxRetries: 5,
+            preflightCommitment: 'confirmed',
+          },
+        );
+
+        // Log for debugging
+        console.log('Transaction sent successfully:', signature);
+
+        // Return signature immediately without waiting for confirmation
+        return signature;
+      } catch (error: unknown) {
+        console.error('TEST Transaction error:', error);
+        if (error instanceof Error) {
+          // Handle insufficient funds error
+          if (error.toString().includes('insufficient lamports')) {
+            throw new Error(
+              `Insufficient balance. Please make sure you have enough SOL to cover the transaction.`,
+            );
+          }
+          // Handle other known errors
+          if (error.toString().includes('Transaction simulation failed')) {
+            throw new Error(`Transaction failed. Please try again.`);
+          }
         }
+        throw error;
       }
-      throw error;
+    } else {
+      console.log('wallet is embedded');
+      const embeddedWallet = wallet as EmbeddedWallet;
+      // TODO: handle embedded wallet transfer
+      return null;
```

Comment:
case 1 is tested and working for all external wallets (privy embedded wallets, external sol wallets like solflare and phantom)

case 2 to enable users to leverage legacy embedded wallets remains to be done. This is not a trivial case since legacy wallets are non privy wallets and hence do not contain the wallet transaction functionality. This needs to leverage the open source solana agent kit. However solana agent kit transfer currently do not support memo. 

Hence I need to modify my fork implementation of solana agent kit to support sending memo.

Reply:
case 2 is implemented and verified in https://github.com/NeurProjects/neur-app/pull/238/commits/3522797f2d57d4d33d76ad38f484737217c12b15 

Will do a final round of testing and a demo vid. Then we can wrap this PR up and merge to `beta`